### PR TITLE
separated code into separate files added initialization code for creatin...

### PR DESCRIPTION
**Pull request by guand**
The changes are as follows:
- Moved code from server.js file to another file 
- added initialization for creating initial data folder on server start up
- JSON files now update if application runs the same repo on a different date instead of not updating if it exists
- added a auth folder for storing tokens

### CODE REVIEW

File: library/directory.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                        callback(null, result);
+                    });
+                });
+            } else {
+                var currentDate = new Date();
+                currentDate.setHours(0, 0, 0, 0, 0);
+                var lastModified = new Date(stats.mtime);
+                lastModified.setHours(0, 0, 0, 0, 0);
+
+                if(currentDate.getTime() == lastModified.getTime()) {
```
Comment (paulmoon) [inline library/directory.js:70]: Is this checking if the file is not older than one hour? That might be too small of a time window, given that # of pull requests/issues don't change much over one day or even a week.

File: scripts/github.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 var GitHubAPI = require('github');
+var auth 	  = require('../auth/auth');
```
Comment (paulmoon) [inline scripts/github.js:2]: Does this work? Shouldn't it say `require('../auth/auth.json')`?

Comment (guand) [inline scripts/github.js:2]: this works for json, might work for js files to but not sure

File: library/directory.js (hunk 1 above)
Comment (guand) [inline library/directory.js:70]: This is checking for a day,
I was thinking this for the hackerNews and Reddit.

Comment (paulmoon) [inline library/directory.js:70]: Okay, sound good.

**Pull request merged**